### PR TITLE
[SPARK-39933][SQL][TESTS] Check query context by `checkError()`

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to check the query context by `checkError()`.

### Why are the changes needed?
To extend test coverage.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
By running the modified test suites:
```
$ build/sbt "test:testOnly *QueryExecutionAnsiErrorsSuite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@gengliangwang @cloud-fan @anchovYu Could you review this PR, please.

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryErrorsSuiteBase.scala [hunk 1]
```diff
@@ -51,4 +51,17 @@ trait QueryErrorsSuiteBase extends SharedSparkSession {
     assert(exception.getSqlState === sqlState)
     assert(exception.getMessage === s"""\n[$fullErrorClass] """ + message)
   }
+
+  case class ExpectedContext(
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryErrorsSuiteBase.scala:55]: why do we need this? can we use `SQLQueryContext` directly in the test?

Comment (MaxGekk) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryErrorsSuiteBase.scala:55]: For simplicity, `ExpectedContext` contains only import info provided via `QueryContext`. For instance, SQLQueryContext contains entire SQL text NOT fragment. Why should we check entire SQL statements in tests?

File: core/src/test/scala/org/apache/spark/SparkFunSuite.scala [hunk 2]
```diff
@@ -318,6 +319,15 @@ abstract class SparkFunSuite
     } else {
       assert(expectedParameters === parameters)
     }
+    val actualQueryContext = exception.getQueryContext()
+    assert(actualQueryContext.length === queryContext.length, "Invalid length of the query context")
+    actualQueryContext.zip(queryContext).foreach { case (actual, expected) =>
```
[the remark this answers is not in the document]
Comment (MaxGekk) [reply core/src/test/scala/org/apache/spark/SparkFunSuite.scala:324]: why? I know exactly which field is invalid in my implementation. In your case, I will only know that contexts are different and then I will have to compare fields by eyes.

**Review by cloud-fan — APPROVED** [review]

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @cloud-fan for review.

**Pull request closed without merging**